### PR TITLE
Validate dev dependencies installation on CI

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [ ] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the section below.

### Changes description
...

### CODE REVIEW

**Review by sfc-gh-pjob — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -78,3 +78,31 @@ jobs:
...
+    needs: lint
+    strategy:
+      matrix:
+          - os: ubuntu-latest
+            TARGET: linux
+            CMD_BUILD: >
+              pyinstaller snowcli.spec &&
+              cd ./dist/snow &&
+              zip -g ../../snow.zip -r .
+            ARTIFACT_NAME: snow-${{ github.sha }}-linux
```
Comment (sfc-gh-astus) [inline .github/workflows/build.yaml:93]: Do we need to keep that artifact?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -78,3 +78,31 @@ jobs:
...
+            PATH: snow.zip
+    steps:
+      - uses: actions/checkout@v3
```
Comment (sfc-gh-astus) [inline .github/workflows/build.yaml:96]: add 

```
with:
          persist-credentials: false
```

**Comment by sfc-gh-turbaszek** [comment]
The build is passing:
https://github.com/sfc-gh-turbaszek/snowcli/actions/runs/5552268954

File: .github/workflows/build.yaml (hunk 1 above)
Comment (sfc-gh-turbaszek) [reply .github/workflows/build.yaml]: Removed

File: .github/workflows/build.yaml (hunk 2 above)
Comment (sfc-gh-turbaszek) [reply .github/workflows/build.yaml:85]: Done

**Review by sfc-gh-astus — APPROVED** [review]

**Review by sfc-gh-jsikorski — APPROVED** [review]

**Pull request merged**